rewriter/Minitest.cc

```diff
@@ -267,25 +312,86 @@ ast::ExpressionPtr runUnderEach(core::MutableContext ctx, core::NameRef eachName
+                                  std::move(moduleName));
+        } else if (insideDescribe &&
+                   ((send->fun == core::Names::let() && send->numPosArgs() == 1) ||
+                    (send->fun == core::Names::letBang() && send->numPosArgs() == 1) ||
+                    (send->fun == core::Names::subject() && send->numPosArgs() <= 1)) &&
+                   correctBlockArity && ast::isa_tree<ast::Literal>(send->getPosArg(0))) {
+            core::NameRef methodName;
+            core::LocOffsets declLoc;
+
+            if (send->numPosArgs() == 1) {
+                auto argLiteral = ast::cast_tree_nonnull<ast::Literal>(send->getPosArg(0));
+                if (argLiteral.isName()) {
+                    declLoc = send->loc.copyWithZeroLength().join(argLiteral.loc);
+                    methodName = argLiteral.asName();
+                } else {
+                    return nullptr;
+                }
+            } else {
+                declLoc = send->loc.copyWithZeroLength().join(send->funLoc);
+                methodName = core::Names::subject();
             }
+
+            // Handle let blocks the same way as it blocks - give them access to test_each arguments
+            ConstantMover constantMover;
+            ast::ExpressionPtr body = move(send->block()->body);
+
+            if (!ast::isa_tree<ast::EmptyTree>(body)) {
+                ast::TreeWalk::apply(ctx, constantMover, body);
+
+                // add the destructuring statements to the block if they're present
+                if (!destructuringStmts.empty()) {
+                    ast::InsSeq::STATS_store stmts;
+                    for (auto &stmt : destructuringStmts) {
+                        stmts.emplace_back(stmt.deepCopy());
+                    }
+                    body = ast::MK::InsSeq(body.loc(), std::move(stmts), std::move(body));
+                }
+            }
+
+            // pull the arg and the iteratee in and synthesize `iterate.each { |arg| body }`
+            ast::MethodDef::ARGS_store new_args;
+            for (auto &arg : args) {
+                new_args.emplace_back(arg.deepCopy());
+            }
+
+            auto blk = ast::MK::Block(send->block()->loc, move(body), std::move(new_args));
+            auto each = ast::MK::Send0Block(send->loc, iteratee.deepCopy(), core::Names::each(),
+                                            send->loc.copyWithZeroLength(), move(blk));
+
+            // let/subject blocks should return their computed value, not void
+            ast::ExpressionPtr method;
+            if (send->fun == core::Names::let() || send->fun == core::Names::letBang() ||
+                send->fun == core::Names::subject()) {
+                method = ast::MK::SyntheticMethod0(send->loc, declLoc, methodName, move(each));
+            } else {
+                method = addSigVoid(ast::MK::SyntheticMethod0(send->loc, declLoc, methodName, move(each)));
+            }
```

Comment:
Isn't the `else` branch dead? how could we get here and not have `send->fun` be one of these three method names?

Reply:
👍🏼 957319d87